homeassistant/components/zwave_js/update.py

```diff
@@ -59,72 +64,75 @@ class ZWaveNodeFirmwareUpdate(UpdateEntity):
 
-    async def async_update(self, write_state: bool = False) -> None:
+    async def _async_update(self, _: HAEvent | datetime | None = None) -> None:
         """Update the entity."""
         for status, event_name in (
             (NodeStatus.ASLEEP, "wake up"),
             (NodeStatus.DEAD, "alive"),
         ):
             if self.node.status == status:
-                if not self._status_unsub:
-                    self._status_unsub = self.node.once(
-                        event_name, self._update_on_status_change
-                    )
-                return
-
-        if available_firmware_updates := (
-            await self.driver.controller.async_get_available_firmware_updates(
-                self.node, API_KEY_FIRMWARE_UPDATE_SERVICE
-            )
-        ):
-            self._latest_version_firmware = max(
-                available_firmware_updates,
-                key=lambda x: AwesomeVersion(x.version),
-            )
-            self._async_process_available_updates(write_state)
+                if self._status_unsub:
+                    return
+                self._status_unsub = self.node.once(
+                    event_name, self._update_on_status_change
+                )
 
-    @callback
-    def _async_process_available_updates(self, write_state: bool = True) -> None:
-        """
-        Process available firmware updates.
-
-        Sets latest version attribute and FirmwareUpdateInfo instance.
-        """
-        # If we have an available firmware update that is a higher version than what's
-        # on the node, we should advertise it, otherwise we are on the latest version
-        if (firmware := self._latest_version_firmware) and AwesomeVersion(
-            firmware.version
-        ) > AwesomeVersion(self.node.firmware_version):
-            self._attr_latest_version = firmware.version
-        else:
-            self._attr_latest_version = self._attr_installed_version
-        if write_state:
-            self.async_write_ha_state()
+        async with self.semaphore:
+            if available_firmware_updates := (
+                await self.driver.controller.async_get_available_firmware_updates(
+                    self.node, API_KEY_FIRMWARE_UPDATE_SERVICE
+                )
+            ):
+                self._latest_version_firmware = max(
+                    available_firmware_updates,
+                    key=lambda x: AwesomeVersion(x.version),
+                )
+                # If we have an available firmware update that is a higher version than
+                # what's on the node, we should advertise it, otherwise we are on the
+                # latest version
+                if (
+                    firmware := self._latest_version_firmware
+                ) is None or AwesomeVersion(firmware.version) <= AwesomeVersion(
+                    self.node.firmware_version
+                ):
+                    return
```

Comment:
Should we not check again after 24 hours for later firmwares?

Reply:
good point, I will revert this change